java/ql/lib/semmle/code/java/frameworks/Strings.qll

```diff
@@ -35,26 +44,43 @@ private class StringSummaryCsv extends SummaryModelCsv {
         "java.lang;String;false;valueOf;(char);;Argument[0];ReturnValue;taint",
         "java.lang;String;false;valueOf;(char[],int,int);;Argument[0];ReturnValue;taint",
         "java.lang;String;false;valueOf;(char[]);;Argument[0];ReturnValue;taint",
+        // TODO: Should `append` and `write` be modelled for Appendable and Writer instead?
+        // Could then remove some of the modelled `append` method here and for StringBuilder
         "java.io;StringWriter;true;append;;;Argument[0];Argument[-1];taint",
         "java.io;StringWriter;true;append;;;Argument[-1];ReturnValue;value",
+        "java.io;StringWriter;true;getBuffer;;;Argument[-1];ReturnValue;taint",
+        "java.io;StringWriter;true;toString;;;Argument[-1];ReturnValue;taint",
         "java.io;StringWriter;true;write;;;Argument[0];Argument[-1];taint",
+        // Note: `AbstractStringBuilder` is a JDK internal superclass of StringBuilder and StringBuffer
+        // Some of the methods are not modelled because they are already modelled for CharSequence
         "java.lang;AbstractStringBuilder;true;AbstractStringBuilder;(String);;Argument[0];Argument[-1];taint",
         "java.lang;AbstractStringBuilder;true;append;;;Argument[0];Argument[-1];taint",
         "java.lang;AbstractStringBuilder;true;append;;;Argument[-1];ReturnValue;value",
+        "java.lang;AbstractStringBuilder;true;appendCodePoint;;;Argument[-1];ReturnValue;value",
+        "java.lang;AbstractStringBuilder;true;delete;;;Argument[-1];ReturnValue;value",
+        "java.lang;AbstractStringBuilder;true;deleteCharAt;;;Argument[-1];ReturnValue;value",
+        "java.lang;AbstractStringBuilder;true;getChars;;;Argument[-1];Argument[2];value",
         "java.lang;AbstractStringBuilder;true;insert;;;Argument[1];Argument[-1];taint",
         "java.lang;AbstractStringBuilder;true;insert;;;Argument[-1];ReturnValue;value",
-        "java.lang;AbstractStringBuilder;true;replace;;;Argument[-1];ReturnValue;value",
         "java.lang;AbstractStringBuilder;true;replace;;;Argument[2];Argument[-1];taint",
-        "java.lang;AbstractStringBuilder;true;toString;;;Argument[-1];ReturnValue;taint",
+        "java.lang;AbstractStringBuilder;true;replace;;;Argument[-1];ReturnValue;value",
+        "java.lang;AbstractStringBuilder;true;reverse;;;Argument[-1];ReturnValue;value",
+        "java.lang;AbstractStringBuilder;true;substring;;;Argument[-1];ReturnValue;value",
         "java.lang;StringBuffer;true;StringBuffer;(CharSequence);;Argument[0];Argument[-1];taint",
         "java.lang;StringBuffer;true;StringBuffer;(String);;Argument[0];Argument[-1];taint",
-        "java.lang;StringBuilder;true;StringBuilder;;;Argument[0];Argument[-1];taint",
-        "java.lang;CharSequence;true;subSequence;;;Argument[-1];ReturnValue;taint"
+        "java.lang;StringBuilder;true;StringBuilder;(CharSequence);;Argument[0];Argument[-1];taint",
+        "java.lang;StringBuilder;true;StringBuilder;(String);;Argument[0];Argument[-1];taint",
```

Comment:
Have specified the parameter types here to avoid tracking `StringBuilder(int)` where the argument only represents the capacitiy.